DNN Platform/Library/Obsolete/HostController.cs

```
@@ -0,0 +1,60 @@
+    {
+        private static IHostController currentInstance;
+
+        [Obsolete("Deprecated in 9.7.1. Scheduled for removal in v11.0.0, use DotNetNuke.Abstractions.Controllers.IHostController instead.")]
+        public static new IHostController Instance
+        {
+            get
+            {
+                if (currentInstance != null)
+                {
+                    return currentInstance;
+                }
+
+                var newHostController = Globals.DependencyProvider.GetRequiredService<INewHostController>();
+                return newHostController is IHostController castedController ? currentInstance = castedController : new HostController();
+            }
+        }
+
+        /// <inheritdoc/>
+        [Obsolete("Deprecated in 9.7.1. Scheduled for removal in v11.0.0, use DotNetNuke.Abstractions.Controllers.IHostController instead.")]
+        public Dictionary<string, ConfigurationSetting> GetSettings() =>
+            (this as INewHostController).GetSettings()
+                .Where(setting => setting.Value is ConfigurationSetting)
+                .Select(setting => new KeyValuePair<string, ConfigurationSetting>(setting.Key, (ConfigurationSetting)setting.Value))
+                .ToDictionary(setting => setting.Key, setting => setting.Value);
+
+        /// <inheritdoc/>
+        [Obsolete("Deprecated in 9.7.1. Scheduled for removal in v11.0.0, use DotNetNuke.Abstractions.Controllers.IHostController instead.")]
+        public void Update(ConfigurationSetting config) =>
+            (this as INewHostController).Update(config);
+
+        /// <inheritdoc/>
+        [Obsolete("Deprecated in 9.7.1. Scheduled for removal in v11.0.0, use DotNetNuke.Abstractions.Controllers.IHostController instead.")]
+        public void Update(ConfigurationSetting config, bool clearCache) =>
+            (this as INewHostController).Update(config, clearCache);
```

I would prefer that this use a regular cast instead of `as`, so that if anything goes wrong, it'll throw an invalid cast exception instead of a null reference exception.

---

Something like this?

```suggestion
            ((INewHostController)this).Update(config, clearCache);
```

---

yup!

---

This has been fixed in the latest code push